exonum/src/encoding/protobuf/mod.rs

```diff
@@ -74,12 +73,13 @@ impl ProtobufConvert for crypto::Hash {
         hash
     }
 
-    fn from_pb(pb: Hash) -> Result<Self, ()> {
+    fn from_pb(pb: Hash) -> Result<Self, Error> {
         let data = pb.get_data();
         if data.len() == crypto::HASH_SIZE {
```

Comment:
You can use a `ensure!` macro which similar to the `assert!` macro but throws error instead of panic.